src/components/timeline/Markers.js

```diff
@@ -27,6 +28,196 @@ export const MIN_MARKER_WIDTH = 0.3;
 
 type MarkerState = 'PRESSED' | 'HOVERED' | 'NONE';
 
+type MouseEventHandler = (SyntheticMouseEvent<HTMLCanvasElement>) => any;
+
+/**
+ * When adding properties to these props, please consider the comment above the component.
+ */
+type CanvasProps = {|
+  +rangeStart: Milliseconds,
+  +rangeEnd: Milliseconds,
+  +width: CssPixels,
+  +height: CssPixels,
+  +getMarker: MarkerIndex => Marker,
+  +markerIndexes: MarkerIndex[],
+  +hoveredItem: Marker | null,
+  +mouseDownItem: Marker | null,
+  +onMouseDown: MouseEventHandler,
+  +onMouseUp: MouseEventHandler,
+  +onMouseMove: MouseEventHandler,
+  +onMouseOut: MouseEventHandler,
+|};
+
+function _drawRoundedRect(
```

Comment:
I extracted this out of the class because it doesn't use anything from the class, mostly when I was looking at which props I had to keep. I can put it back if you prefer.

Reply:
Works for me!